src/crawlers/playwright_crawler.js

```diff
@@ -299,6 +299,42 @@ class PlaywrightCrawler extends BrowserCrawler {
         this.launchContext = launchContext;
     }
 
+    /**
+     * Returns context options that will be used when a page needs to be created.
+     * Required for proxy per page.
+     *
+     * @param {BrowserCrawlingContext & CrawlingContext} crawlingContext
+     * @ignore
+     * @protected
+     * @internal
+     */
+    _getNewPageOptions(crawlingContext) {
+        if (this.proxyConfiguration && this.launchContext.useIncognitoPages) {
+            const pageOptions = {};
+
+            const { session } = crawlingContext;
+            const { url } = this.proxyConfiguration.newProxyInfo(session && session.id);
+
+            const parsed = new URL(url);
+
+            pageOptions.proxy = {
+                server: parsed.origin,
+                username: parsed.username,
+                password: parsed.password,
```

Comment:
Sorry, it only occured to me now, but `launchContext` supports `proxyUrl` and therefore encapsulates the internal proxy configuration of Puppeteer and Playwright. I think we should do it the same way for setting a proxy in the new page options to stay consistent.

I suggest adding `proxyUrl` as another argument of new page options, next to `id` in `browser-pool`. As a syntax sugar for having to know this proxy configuration in both Puppeteer and Playwright. It will simplify and unify proxy setup in `browser-pool` and save us some lines of code here.